Add org.supertux.SuperTux-Milestone1

Grumbel opened the pull request:
This is the classic SuperMarioBros1 inspired version of SuperTux, which is separate from the already existing `org.supertuxproject.SuperTux`, which is the modern one.

I am using the proper `org.supertux` prefix here, `org.supertuxproject` should be renamed at some point (will fill a separate issue for that).

Alexander-Wilms commented:
The aarch64 build failed: https://flathub.org/builds/#/builders/2/builds/2891

nedrichards commented:
Huh, how did that fail? It's a shared module that already has the reconfigure stuff in it. https://github.com/flathub/shared-modules/blob/master/SDL/SDL_mixer-1.2.12.json

Grumbel commented:
I am not using the flathub/shared-modules version of the SDL_mixer-1.2.12.json module as that one failed with this error at the configure stage:

```
checking signal.h usability... yes
checking signal.h presence... yes
checking for signal.h... yes
checking for setbuf... yes
./configure: line 13698: syntax error near unexpected token `$SMPEG_VERSION,'
./configure: line 13698: `    AM_PATH_SMPEG($SMPEG_VERSION, have_smpeg=yes, have_smpeg=no)'
Error: module SDL_mixer: Child process exited with code 2
```

The version I am using is just the original tarball with the original autotools stuff still in it, which is probably to old judging by the build-bots error message.

nedrichards commented:
Oh, right - yeah. You need the reconfigure stuff that I linked above so that it learns about aarch64.

nedrichards commented:
bot, build org.supertux.SuperTux-Milestone1

Grumbel commented:
I added "shared-modules/smpeg/smpeg-0.4.5.json" to the modules and switched back to the SDL_mixer module from flathub/shared-modules, this fixes the SMPEG issue I had. 

The SDL_Mixer module should probably do a `autoconf -I acinclude/`.

nedrichards commented:
Just to check, since I haven't built this locally or investigated yet - will these two be visually differentiated in app stores?

nedrichards commented:
Yeah, makes sense. Patches welcome ;-)

Grumbel commented:
The program is named "SuperTux - Milestone 1", the .svg icon has a different background color and all the screenshots are in their original 640x480 glory. Shouldn't be to hard to tell apart.

nedrichards commented:
Hah! Sounds good. The screenshots in that rez may cause a few issues but there's a couple of other apps here that just have native resolutions that are smaller than recommended so it'll work out.

nedrichards commented:
Repository has been created: https://github.com/flathub/org.supertux.SuperTux-Milestone1
